test/runnable/iasm.d

```diff
@@ -4786,6 +4786,18 @@ void test5012()
 
 /****************************************************/
 
+import core.vararg;
+extern(C) void func16400(int, ...) {
+    asm {naked; ret;};
+}
+
+void test16400()
+{
+    assert(*(cast(ubyte*) &func16400) == 0xc3); // ret
```

Comment:
It doesn't work with the `D` linkage because it uses a different technique/code path.
I think that's a non-issue since if the user chooses the `D` linkage he's expected to deal with the typeinfo and the `_arguments` parameter; another possibility is emitting an error if the user tries to have a naked variadic function with D linkage.